server/agent/views.py

```diff
@@ -193,3 +195,16 @@ class AvailableAgentsView(APIView):
     )
     def get(self, request):
         return Response({"choices": settings.AVAILABLE_AGENTS.keys()}, status=status.HTTP_200_OK)
+
+
+class ConfigView(APIView):
+    permission_classes = [IsAuthenticated]
+    """View to get all available configuration options (nested) with serializer."""
```

Comment:
I'm not sure if I understand what this API endpoint does - what are the configuration options that it exposes?

Reply:
It returns all possible classes(registered in settings.py) for agent config. e.g. Agent, Tools, Retrievers etc.

Reply:
It's quite a low level, but in the end configuration will be set by admins right?

I was thinking about it from dev perspective. Let's say I've added new retriever which needs a custom repo and I need to compare results. At this point I can setup some tests easily, without any code, except adding new classes to Settings, and Agent is not strictly tied up to retriever.

Also, I didn't wanted to split config into API + in code one
If you feel, that hiding some of configuration items from API and do it in code may be a better experience, we can go this way